src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestState.cs

```diff
@@ -55,6 +55,24 @@ public object Lock
             }
         }
 
+        public void ClearSendRequestState()
+        {
+            // Since WinHttpRequestState has a self-referenced strong GCHandle, we
```

Comment:
Would there be any value in restructuring the object so that the finalizable thing and the GCHandle both point to the same sub-object, in a manner that doesn't have any cycles?